feat: 메인페이지 UI 구현 (#36)

jngmnj opened the pull request:
## 🔗 관련 이슈
- #36
- #67 

## ✨ 작업 개요
- 메인 페이지를 다양한 섹션으로 구성하고 관련된 컴포넌트들을 구현했습니다.
- Next.js의 App Router 환경에서 발생한 빌드 오류를 수정하고, 컴포넌트 구조를 개선했습니다.
- UI/UX 개선을 위해 캐러셀, 스켈레톤, 반응형 레이아웃 등을 적용했습니다.

## ✅ 작업 상세 내용
- **메인 페이지 섹션 추가 및 컴포넌트 구현**
  - 메인 배너 캐러셀(MainBannerCarousel), 추천 게임 섹션(RecommendedGameSection), NEW & HOT 게임 섹션(NewAndHotGameSection), 지금 뜨는 리뷰 섹션(TrendingReviewSection)을 메인 페이지에 추가했습니다.
  - 각 섹션에 필요한 `Wrapper` 및 `Item` 컴포넌트들을 구현하고, `Swiper` 라이브러리를 활용한 캐러셀 기능을 적용했습니다.
  - 로그인 상태에 따라 추천 게임 섹션 또는 취향 설문 CTA(`PreferenceSurveyCTA`)가 노출되도록 분기 처리했습니다.
  - `Suspense`와 스켈레톤 UI를 도입하여 로딩 중 사용자 경험을 개선했습니다.

- **타입 및 데이터 리팩터링**
  - 기존 `Game` 타입을 `GameData`로 통합하고, `image_url` 필드를 `thumbnail_url`로 변경했습니다.
  - 게임의 플레이 타임을 `playtime_minutes`에서 `playtime_min_minutes`, `playtime_max_minutes`로 분리하여 데이터의 정확성을 높였습니다.
  - `user` mock 데이터를 별도의 파일(`userProfile.ts`)로 분리하여 컴포넌트의 의존성을 낮추고 빌드 오류를 해결했습니다.
  - `review` 데이터 타입에 `game_id`, `title`, `image_url` 필드를 추가하여 관련된 컴포넌트들을 수정했습니다.

- **UI/UX 및 버그 수정**
  - `src/app/page.tsx`의 루트 요소를 `<>`에서 `<main>`으로 변경하여 semantic HTML을 준수하도록 했습니다.
  - 캐러셀 네비게이션 버튼의 클릭이 작동하지 않는 문제를 해결하기 위해 `pointer-events` 속성을 조정하고, `div`를 `button`으로 변경하여 접근성을 개선했습니다.
  - RankItem 컴포넌트의 이미지 렌더링 방식을 최적화하여 반응형 및 로딩 성능을 개선했습니다.
  - 모바일 레이아웃 및 스타일을 조정하여 사용성을 향상했습니다.

## ✨ 주요 기능
- **메인 페이지 섹션별 캐러셀 및 레이아웃**
  - 메인 배너, 추천 게임, NEW & HOT 게임, 지금 뜨는 리뷰 섹션 제공
  - `Swiper`를 활용한 자동 슬라이드 및 반응형 캐러셀 구현
- **로그인 상태별 UI 분기**
  - 로그인 유저는 추천 게임 리스트 확인 가능
  - 비로그인 유저에게는 취향 설문 CTA 노출
- **랭킹 및 리뷰 리스트 컴포넌트**
  - `RankItemList`를 통해 NEW & HOT 게임 순위 시각적으로 강조
  - `ReviewItem`을 통해 다양한 리뷰 정보를 유연하게 표시
- **빌드 오류 수정 및 코드 구조 개선**
  - `ReferenceError` 해결 및 mock 데이터 분리를 통한 유지보수성 향상
  - `Game` 관련 타입 및 데이터 구조 통일
  
  ## 📸 UI 스크린샷
<img alt="localhost_3000_ (1)" src="https://github.com/user-attachments/assets/c2d1941a-756b-46a3-8660-ef7c5959e33d"  />

jngmnj committed:
feat: 메인 배너 캐러셀 컴포넌트 추가 및 목데이터 적용 (#36)

- MainBannerCarousel 및 Wrapper 컴포넌트 구현
- boardlife.co.kr 이미지 사용을 위한 이미지 도메인 및 remotePatterns 설정 추가
- 메인 배너용 mock 데이터(mainBannerList.ts) 작성
- Swiper 기반 자동 슬라이드 및 반응형 레이아웃 구현

jngmnj committed:
chore: 메인페이지 이미지 추가 (#36)

jngmnj committed:
refactor: Game 타입을 GameData로 통일 및 thumbnail_url 필드로 변경 (#36)

- GameListData 및 Game 타입을 GameData로 통합
- 기존 image_url 필드를 thumbnail_url로 변경 및 관련 컴포넌트 수정
- 게임 상세 페이지에서 playtime_minutes → playtime_min/max_minutes로 분리 적용
- GameItem, GameList, GameDetailInfo, SimilarGameList 등 관련 컴포넌트 일괄 수정
- mock 데이터(gameListData.ts) 구조 개선 및 게임 항목 추가

jngmnj committed:
feat: 메인 페이지에 배너 캐러셀 및 게임 테마 큐레이션 섹션 추가 (#36)

- MainBannerCarouselWrapper 컴포넌트 메인 페이지에 렌더링
- GameThemeCuration 컴포넌트 신규 추가 및 입문자/2인용/파티게임/전략가 큐레이션 카드 구성
- theme별 게임 리스트로 이동 가능한 링크 연결
- 반응형 레이아웃 및 스타일 적용

jngmnj committed:
feat: 추천 게임 캐러셀 섹션 및 취향 설문 CTA 추가 (#36)

- 로그인 유저 기반 추천 게임 섹션(RecommendedGameList) 추가
- Swiper 슬라이더 적용 및 반응형 레이아웃 구성
- 유저 정보 없을 경우 PreferenceSurveyCTA 안내 배너 노출
- Wrapper 컴포넌트(RecommendedGameListWrapper) 및 전역 버튼 스타일(theme-button-next/prev) 정의
- 메인 페이지에 추천 게임 섹션 연결

jngmnj committed:
refactor: 추천 게임 리스트 컴포넌트 구조 개선 및 스켈레톤 추가 (#36)

- RecommendedGameList를 props 기반 컴포넌트로 리팩터링
- Suspense를 활용한 RecommendedGameListWrapper 구현
- 로딩 시 보여줄 RecommendedGameListSkeleton 컴포넌트 추가
- 사용자 로그인 여부 분기 및 CTA를 포함한 RecommendedGameSection 분리

jngmnj committed:
feat: NEW & HOT 게임 섹션 및 랭킹 리스트 컴포넌트 추가 (#36)

- NewAndHotGameSection 컴포넌트를 메인 페이지에 적용
- RankItemList, RankItem 컴포넌트를 분리하여 게임 랭킹 리스트 구현
- 상위 3개 게임을 NEW, HOT 섹션으로 나누어 시각적으로 강조
- page.tsx에서 RecommendedGameSection 및 NewAndHotGameSection 통합 관리

jngmnj committed:
chore: ReviewItem 타입 수정 및 기본 이미지 추가 (#36)

- game_id, title을 필수 속성으로 변경
- image_url 속성 추가
- 리뷰 이미지 없을 경우 사용할 기본 이미지(img_default.png) 추가

jngmnj committed:
feat: 리뷰 카드 UI를 ReviewItem 컴포넌트로 분리 (#36)

- ReviewItem 컴포넌트를 생성하여 다른 사용자 리뷰와 내 리뷰를 조건부 렌더링
- 이미지, 제목, 날짜 노출 여부를 props로 제어 가능하도록 구현

jngmnj committed:
feat: 지금 뜨는 리뷰 섹션 및 캐러셀 추가 (#36)

- TrendingReviewSection, TrendingReviewWrapper, TrendingReviewCarousel 컴포넌트 추가
- Swiper를 사용해 모바일에서는 세로, 데스크톱에서는 가로 슬라이드 구현
- 리뷰 클릭 시 ReviewModal 호출

jngmnj committed:
chore: reviewData에 game_id, title, image_url 속성 및 테스트 데이터 추가 (#36)

- 기존 리뷰 mock 데이터에 game_id, title, image_url 필드를 추가하여 타입 변경 사항 반영
- 일부 리뷰 데이터에 기본 게임 이미지 URL 삽입
- 새로운 리뷰 항목(5, 6번) 추가하여 캐러셀/리스트 테스트 데이터 확충

jngmnj committed:
chore: 메인 랭킹/신작·인기 게임 섹션 레이아웃 및 이미지 렌더링 개선 (#36)

- NewAndHotGameSection: 모바일에서 1열(grid-cols-1)로 변경, gap-15 적용 / md 이상에서는 2열 및 gap-6 유지
- RankItem: 이미지 컨테이너에 w-auto 추가로 비율 깨짐 방지, GameMeta의 isLink=false로 변경하여 클릭 범위 제어
- RankItemList: 이미지 렌더링을 width/height 고정에서 fill+priority로 변경해 반응형 대응 및 로딩 성능 개선, h-55 고정 높이 적용

jngmnj committed:
feat: 메인 페이지에 TrendingReviewSection 추가 (#36)

- page.tsx에 TrendingReviewSection 컴포넌트를 임포트 및 렌더링하여 메인 페이지에서 최신 인기 리뷰 섹션 노출

jngmnj committed:
fix: 슬라이드 네비게이션 클릭 불가 문제 수정 (#36)

- 네비게이션 래퍼에 `pointer-events-none` 적용 후 버튼에 `pointer-events-auto` 부여
- 네비게이션 요소를 div → button으로 변경해 접근성 및 클릭 이슈 개선

jngmnj committed:
fix: 메인 페이지 빌드 오류 수정 및 컴포넌트 구조 개선 (#36)

- 빌드 과정에서 `ReferenceError: user is not defined` 오류가 발생하던 문제 해결
- `RecommendedGameSection` 컴포넌트 내부에서 mock user 데이터를 직접 선언하는 대신, 별도의 mock 파일(`src/assets/mocks/userProfile.ts`)로 분리하여 관리하도록 구조 개선
- `src/app/page.tsx`의 루트 요소를 `<>`에서 `<main>`으로 변경하여 semantic HTML을 준수하도록 수정

jngmnj committed:
chore: 메인페이지 마지막섹션 margin bottom 추가 (#36)

devjaesung commented on src/components/game/detail/ReviewItem.tsx:
```diff
@@ -0,0 +1,101 @@
+          <div className="flex items-center gap-3">
+            <Image
+              src={
+                review.user.profile_image_url || '/images/defaultProfileImg.png'
```
기본이미지 같은 경우는 상수로 관리하면 다른 페이지(예: mypage)에서도 쉽게 사용할 수 있을 것 같아요!

devjaesung commented on src/components/main/GameThemeCuration.tsx:
```diff
@@ -0,0 +1,62 @@
+          <Link key={index} href={content.link} className="cursor-pointer">
+            <div className="hover:border-primary-100 hover:drop-shadow-primary-200 flex items-center gap-4 rounded-2xl border border-transparent bg-white px-8 py-4 drop-shadow-xl drop-shadow-transparent transition-all md:flex-col md:flex-wrap md:gap-0 md:rounded-4xl">
+              {' '}
+              <Image
```
저장하다가 공백 들어간것 같아요! 확인해주세요!

jngmnj replied:
매의눈 jaesunggo 감사합니다!

jngmnj committed:
refactor: 기본 이미지 경로 상수화 및 관련 컴포넌트 적용 (#36)

- `/images/defaultProfileImg.png` → DEFAULT_PROFILE_IMAGE 상수로 변경
- `/images/img_default.png` → DEFAULT_IMAGE 상수로 변경
- `src/constants/image.ts`에 기본 이미지 경로 상수 정의
- MyPage, ProfileEditPage, ReviewItem, ReviewModal, ProfileDetail 등 관련 컴포넌트에 상수 적용
- 불필요한 공백 제거
